### PR TITLE
Partner Portal: Refactor key selection to use the http() utilities and to look better

### PR DESCRIPTION
**Pull request by atanas-dev**
Context:
* Project thread: pbtFPC-Um-p2
* i4 designs: pbtFPC-103-p2

#### Changes proposed in this Pull Request

* In the Partner Portal, improve partner key selection UI visually.
* In the Partner Portal, refactor key selection data layer to use the http() utilities.

#### Testing instructions

* If you do not have a partner key, please contact Infinity for one or you will be unable to view the UI.
* Checkout PR locally, run `yarn && yarn start-jetpack-cloud`.
* Visit http://jetpack.cloud.localhost:3000/partner-portal and select your partner key, if prompted.
* Make sure you are shown the error message (shown at the bottom) if you have no active partner keys.
* Make sure key selection shows up if you have multiple active partner keys.
* Make sure key selection is skipped if you have a single active partner key (it should get autoselected).

Key selection:
Screenshot 2021-03-03 at 21 15 22

Missing/error-ed out key selection:
Screenshot 2021-03-03 at 21 17 47
- This screen will show up whenever keys cannot be loaded (e.g. user is not a partner, does not have keys or request failed).
- The error toast message will only appear when the request failed.

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/select-partner-key/index.tsx [hunk 1]
```diff
@@ -44,29 +49,37 @@ export default function SelectPartnerKey(): ReactElement | null {
...
+			<CardHeading size={ 36 }>{ translate( 'Partner Portal' ) }</CardHeading>
 
-				{ ! isFetching &&
-					keys.map( ( key ) => (
-						<Card key={ key.id } className="select-partner-key__card">
+			{ isFetching && <Spinner /> }
+
+			{ showError && (
+				<div className="select-partner-key__error">
+					<p>{ translate( 'Your account is not registered as a partner account.' ) }</p>
```
Comment (IanRamosC) [inline client/jetpack-cloud/sections/partner-portal/select-partner-key/index.tsx:61]: I'm wondering if we should mention how they can get a partner key. If someone gets to this page without a key, they are likely to know about the Partner Portal.

**Review by IanRamosC — APPROVED** [review]
Thanks for the screenshots! Everything works as expected.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5559065

Thank you @atanas-dev for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5559065).

**Pull request merged**